### PR TITLE
e2e: better method of adding SDK repo replace directive to operator's go.mod

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** directly add `replace` directive for the local SDK repo to `go.mod` bytes instead of doing a bunch of extra work to use `go mod edit`.

**Motivation for the change:** other dependencies may have unresolved tags/commits, ex. `kubernetes-1.14.1`, in a newly-scaffolded `go.mod` which cause errors during the parse step of `go mod edit`. Instead of doing extra replace work with `sed` or Go, we can skip using `go mod` entirely and just add a `replace` directive for the SDK repo by modifying `go.mod` bytes directly.

### CODE REVIEW

File: hack/lib/test_lib.sh [hunk 1]
```diff
@@ -41,3 +41,28 @@ function trap_add() {
...
+	if [[ ! -e go.mod ]]; then
+		echo "go.mod file not found in $(pwd)"
+	fi
+
+	# Use the local operator-sdk directory as the repo. To make the go toolchain
+	# happy, the directory needs a `go.mod` file that specifies the module name,
+	# so we need this temporary hack until we update the SDK repo itself to use
+	# Go modules.
+	echo "module ${from_path}" > "${to_path}/go.mod"
+	trap_add "rm ${to_path}/go.mod" EXIT
```
Comment (joelanford) [inline hack/lib/test_lib.sh]: Could we check to see if `$(to_path}/go.mod` exists, and only do these two steps if it doesn't?

Same comment for the go version of this code as well.

Comment (estroz) [reply hack/lib/test_lib.sh]: Added a check for `to_path`. `version` is a repo version so the go toolchain will deal with a non-existent version.

File: hack/lib/test_lib.sh [hunk 2]
```diff
@@ -41,3 +41,30 @@ function trap_add() {
...
+	local to_path="$2"
+	local version="${3:-}"
+
+	if [[ ! -e go.mod ]]; then
+		echo "go.mod file not found in $(pwd)"
+	fi
+
+	# Use the local operator-sdk directory as the repo. To make the go toolchain
+	# happy, the directory needs a `go.mod` file that specifies the module name,
+	# so we need this temporary hack until we update the SDK repo itself to use
```
Comment (hasbro17) [inline hack/lib/test_lib.sh]: Nit: Format this as an explicit TODO comment so it's easier to parse later.

File: test/e2e/memcached_test.go [hunk 3]
```diff
@@ -120,30 +118,21 @@ func TestMemcached(t *testing.T) {
...
 			// the local SDK repo as a module.
 			sdkModPath := filepath.Join(filepath.FromSlash(replace.repo), "go.mod")
-			err = ioutil.WriteFile(sdkModPath, []byte("module "+sdkRepo), fileutil.DefaultFileMode)
-			if err != nil {
-				t.Fatalf("Failed to write main repo go.mod file: %v", err)
-			}
-			defer func() {
-				if err = os.RemoveAll(sdkModPath); err != nil {
-					t.Fatalf("Failed to remove %s: %v", sdkModPath, err)
+			if _, err = os.Stat(sdkModPath); err != nil && os.IsNotExist(err) {
```
Comment (hasbro17) [inline test/e2e/memcached_test.go]: What's supposed to happen if `err != nil && os.IsNotExist(err) == false`, i.e some other error (permissions, filesystem)?
Looking down it seems like `insertGoModReplace()` will fail when it tries to read the file so I guess it's fine.

Comment (estroz) [reply test/e2e/memcached_test.go]: Yep I figured I'd let `ioutil.ReadFile()` handle that error.

**Pull request merged**